spec/requests/api/article_index_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+RSpec.describe 'Article index', type: :request do
+  describe 'GET /api/articles ' do
+    context 'basic specs' do
+      let!(:published_articles) { 5.times { create(:article, published: true) } }
+      let!(:unpublished_articles) { 5.times { create(:article, published: false) } }
+
+      before do
+        get '/api/articles'
+			end
+			
+      it 'returns 200' do
+        expect(response).to have_http_status(200)
+      end
+
+    	it 'returns 5 articles' do
+        expect(JSON.parse(response.body)['articles'].count).to eq 5
+      end
+    end
+
+    context 'detailed spec to test associations etc' do
+      let(:user) { create(:user, email: 'journalist@mail.com') }
+      let(:category) { create(:category, name: 'Sports') }
+      let!(:published_article) { create(:article, title: 'My first article', user: user, category: category, published: true )}
+      let!(:unpublished_article) { create(:article, title: 'My second article', user: user, category: category, published: false  )}
+
+
+      before do
+        get '/api/articles'
+      end
+
+      it 'includes info about user' do
+        expected_email = JSON.parse(response.body)['articles'][0]['user']['email']
+        expect(expected_email).to eq 'journalist@mail.com' 
+      end
+
+      it 'includes info about category' do
+        expected_email = JSON.parse(response.body)['articles'][0]['category']['name']
+        expect(expected_email).to eq 'Sports' 
+      end
+
+      it 'includes info about image' do
+        expected_url = JSON.parse(response.body)['articles'][0]['image']
+        expect(expected_url).to match(/attachment.png/)
+      end
+    end
+  end
+end
```

Comment:
Please mind code formatting here

Reply:
I would also add a test for title

Reply:
Thanks, we do the test for that. Could you tell us what should we change with the code formatting, what is the convention? It looks fine for us in the editor (it looks different in GH)
![screen shot 2019-01-22 at 20 17 08](https://user-images.githubusercontent.com/43814981/51559680-d6cf4380-1e82-11e9-9aee-5778daf1ca3c.png)

Reply:
Consistency is the key here. I have no opinion in the ever ongoing discussion about tabs vs spaces.